addons/account/models/product.py

```diff
@@ -53,6 +54,12 @@ def get_product_accounts(self, fiscal_pos=None):
             fiscal_pos = self.env['account.fiscal.position']
         return fiscal_pos.map_accounts(accounts)
 
+    @api.constrains('uom_id')
+    def _check_uom_not_in_invoice(self):
+        for product in self:
+            invoices = self.env['account.move.line'].sudo().search([('product_id.product_tmpl_id.id', '=', product.id)])
+            if invoices:
+                raise ValidationError(_('The product "%s" is used in invoices. You cannot change the Unit of Measure of it.', self.display_name))
```

Comment:
```suggestion
        for template in self:
            invoices = self.env['account.move.line'].sudo().search([('product_id.product_tmpl_id.id', '=', template.id)], limit=1)
            if invoices:
                raise ValidationError(_('The product "%s" is used in invoices. You cannot change its Unit of Measure.', self.display_name))
```
Since we are dealing with product templates, it might be clearer to rename your variable ?
Adding a limit might speed up the process a bit in some cases :wink:

Reply:
Yep, I was focused on the use case. I make the change!
Thanks for the review :)